### PR TITLE
app_voicemail: Use ast_asprintf to create mailbox SQL query

### PR DESCRIPTION
**Pull request by gtjoseph**
...instead of trying to calculate the length of the buffer needed
manually.

### CODE REVIEW

**Comment by gtjoseph** [comment]
cherry-pick-to: 21
cherry-pick-to: 22

File: apps/app_voicemail.c [hunk 1]
```diff
@@ -6261,12 +6261,14 @@ static int messagecount(const char *mailbox_id, const char *folder)
 	}
 
 	if (!strcmp(folder, "INBOX")) {
-		gps.sql = ast_alloca(sizeof(MSGCOUNT_SQL_FMT_INBOX) + odbc_table_len + (strlen(VM_SPOOL_DIR) + strlen(context) + strlen(mailbox) * 2));
-		sprintf(gps.sql, MSGCOUNT_SQL_FMT_INBOX, odbc_table, VM_SPOOL_DIR, context, mailbox, VM_SPOOL_DIR, context, mailbox); /* Safe */
+		res = ast_asprintf(&gps.sql, MSGCOUNT_SQL_FMT_INBOX, odbc_table, VM_SPOOL_DIR, context, mailbox, VM_SPOOL_DIR, context, mailbox); /* Safe */
```
Comment (seanbright) [inline apps/app_voicemail.c:6264]: `/* Safe */` seems unnecessary now.

File: apps/app_voicemail.c [hunk 2]
```diff
@@ -6261,12 +6261,14 @@ static int messagecount(const char *mailbox_id, const char *folder)
...
 	} else {
-		gps.sql = ast_alloca(sizeof(MSGCOUNT_SQL_FMT) + odbc_table_len + strlen(VM_SPOOL_DIR) + strlen(context) + strlen(mailbox) + strlen(folder));
-		sprintf(gps.sql, MSGCOUNT_SQL_FMT, odbc_table, VM_SPOOL_DIR, context, mailbox, folder); /* Safe */
+		res = ast_asprintf(&gps.sql, MSGCOUNT_SQL_FMT, odbc_table, VM_SPOOL_DIR, context, mailbox, folder); /* Safe */
```
Comment (seanbright) [inline apps/app_voicemail.c:6266]: Idem

File: apps/app_voicemail.c (hunk 1 above)
Comment (gtjoseph) [reply apps/app_voicemail.c]: Oops, yep.

File: apps/app_voicemail.c (hunk 2 above)
Comment (gtjoseph) [reply apps/app_voicemail.c]: Idem

**Review by jcolp — APPROVED** [review]

**Pull request merged**